### DISCUSSION TITLE
Web components with PrimeVue 4, how to include scripts and stylings?

### DISCUSSION THREAD

**Discussion by AlexanderSchmid** [primefaces/community#3081] [PrimeVue]
I'm using PrimeVue 4 with nuxt 3. I've set up our solution as described here: https://primevue.org/nuxt/ 
I configured a custom theme based on Aura in a separate typescript file and imported it with "importTheme" in nuxt.config.ts. That is fine and it works.

I want to generate a custom web component (with nuxt-custom-elements) and my vue component is using primevue components and stylings from our theme. The web component should be used in another application with different technologies.

How can I include the stylings from PrimeVue 4 so that the look and feel is like in our solution?

I did not find any docs. I tried something like that but no success:

`import '#primevue-style';
import 'primeicons/primeicons.css';
import '@primevue/themes';
import 'tailwindcss-primeui';
import '~/assets/css/theme/PrimeVueTheme';`

Can anybody help me? Is it possible to generate PrimeVue 4 as web component? If yes, how?

**Comment by garethjamesmoneyalive** [comment]
Also interested in a solution to this. It appears that PrimeVue 4 doesn't come bundled with any pre-rendered CSS making exporting to web components or a SFC component library very difficult. It seems that we either need a way to trigger the building of CSS files on a per-component basis or the compiled CSS needs to be bundled with each component ready for import into a Vue component.

**Comment by montgomery1944** [comment]
@AlexanderSchmid, support for custom elements was implemented and published in a fork here https://www.npmjs.com/search?q=@primeuix-ce and here https://www.npmjs.com/search?q=@primevue-ce.

Fork will be kept up to date with the upstream repositories.

**Reply by stephanpelikan** [reply]
We develop micro-frontends loaded by module federation into a host-application as web-component for proper isolation. We will use `primevue-ce` for now. Unfortunately, this is not public on Github. We are grateful for having `primevue-ce` but we would appreciate to get support for ce by primevue itself.
